### DISCUSSION TITLE
Connecting a ILI9488 panel directly to RP2040/ESP32 via FPC module: is it possible or do I need a custom PCB?

### DISCUSSION THREAD

**Discussion by Jikeh** [Bodmer/TFT_eSPI#3513] [Q&A - General] [answered]
Hello, noob here.

I've recently acquired one of this ILI9488 ISP bare panels (https://aliexpress.com/item/1005006487663199.html) with the objective driving it via 8/16bit parallel either with a RP2040 or an ESP32-S3. The problem is that there is no way to get it to work. I had previously used a very similar display that comes with a PCB which exposes SPI pins (https://aliexpress.com/item/1005003033844928.html), which works just fine, but I want faster drawing speed.

I have confirmed that the new ISP panel works if i connect the ribbon to the red PCB module, but since that red PCB only exposes SPI pins I can't use it for parallel. I bought a small FPC 40pin 0.5mm module (https://aliexpress.com/item/1005007081601754.html) that takes the ribbon from the panel and exposes all 40 pins and connected them such as:

```
// RP2040                   |       TFT (fpc 40)
// 21 (GP16/SPI0_RX/MISO)   ->      14 (SPI_SDO)
// 22 (GP17/SPI0_CS)        ->      9 (SPI CS)
// 23 (GND)                 ->      5 (GND)
// 24 (GP18/SPI0_SCK)       ->      11 (SPI_SCL)
// 25 (GP19/SPI0_TX/MOSI)   ->      13 (SPI_SDI/SDA)
// 26 (GP20)                ->      10 (DC)
// 27 (GP21)                ->      15 (RESET)
// 29 (GP22)                ->      33 (LED-A (BL))
// 28 (GND)                 ->      16 (GND)
// 36 (3V3)                 ->      6 (VCC)
// 36 (3V3)                 ->      7 (VCI)
// 13 (GND)                 ->      34 (GND)
// 18 (GND)                 ->      35 (GND)
// 18 (GND)                 ->      36 (GND)
// 3 (GND)                  ->      37 (GND)
```

Therefore setting:
```

#define TFT_MISO 16
#define TFT_MOSI 19
#define TFT_SCLK 18
#define TFT_CS   17
#define TFT_DC   20 
#define TFT_RST   21 
#define TFT_BL   22   

#define TFT_SPI_PORT 0 // Set to 0 if SPI0 pins are used, or 1 if spi1 pins used
#define TFT_BACKLIGHT_ON HIGH  // Level to turn ON back-light (HIGH or LOW)
```

And here are the pinouts (Panel on the left, RP2040 on the right):
![photo_2024-10-18_11-16-37](https://github.com/user-attachments/assets/5e14a3e8-295c-4530-887a-912d8a884453)

Should be enough to use SPI (testing SPI first since it requires less wires)

However the panels does does not turn on at all. After some research it occured to me that it might not be possible to connect the pins of the ribbon directly to the MCU and I need instead a custom PCB such as the red one in my second link. 

Is my intuition correct or is it in fact possible to connect the TFT panel directly to the MCU? If so, what am I doing wrong?

**Comment by Basitadam** [comment, marked as answer]
Hi,
If you driving backlight directly from IO pins, it won't lit. Product page says needs 120mA. Try driving via a FET or a transistor
